crypto/init.c

```diff
@@ -684,7 +706,47 @@ int OPENSSL_atexit(void (*handler)(void))
     return 1;
 }
 
-#ifdef OPENSSL_SYS_UNIX
+#ifndef OPENSSL_SYS_UNIX
+CRYPTO_RWLOCK *CRYPTO_THREAD_glock_new(const char *name)
+{
+    return CRYPTO_THREAD_lock_new();
+}
+#else
+
+
+/*
+ * Create a new global lock, return NULL on error.
+ */
+CRYPTO_RWLOCK *CRYPTO_THREAD_glock_new(const char *name)
+{
+    GLOBAL_LOCK *newlock;
+
+    if (name == NULL
+            || glock_lock == NULL
+            || (newlock = malloc(sizeof(*newlock))) == NULL)
+        return CRYPTO_THREAD_lock_new();
```

Comment:
It seems like it might be better to take advantage of the documented error return here than try some unpredictable fallback scheme.

Reply:
This stuff gets called during auto-init which really doesn't have a way to indicate errors to the application.  I tweaked the comment.  I also changed the if test to only check malloc().

Reply:
and created the glock lock via run-once now.